lib/heartcheck/webservice/http_client.rb

```diff
@@ -3,17 +3,29 @@ module Heartcheck
   module Webservice
     # Http Client
     class HttpClient
-      attr_reader :uri, :http
+      attr_reader :uri, :http, :proxy, :headers, :request
 
-      def initialize(url, ignore_ssl_cert)
+      def initialize(url, proxy, ignore_ssl_cert, headers)
         @uri = URI url
-        @http =  Net::HTTP.new(@uri.host, @uri.port)
+        @headers = headers || {}
+
+        if proxy
+          @proxy = URI proxy
+          @http = Net::HTTP.new(@uri.host, @uri.port, @proxy.host, @proxy.port)
+        else
+          @http =  Net::HTTP.new(@uri.host, @uri.port)
+        end
```

Comment:
In my opinion, we should keep the initialize method as short as possible. Due to this what do you think to do something like that:

```ruby
  def initialize(url, proxy, ignore_ssl_cert, headers)
    @uri = URI url
    @proxy = URI(proxy) if proxy.present?
    @headers = headers || {}
    @http =  Net::HTTP.new(@uri.host, @uri.port, proxy_host, proxy_port)
    @http.use_ssl = @uri.scheme == 'https'
    @http.verify_mode = OpenSSL::SSL::VERIFY_NONE if ignore_ssl_cert
  end

  private

  def proxy_port
    proxy.port if proxy
  end

  def proxy_host
    proxy.host if proxy
  end
```

We do not need to check if there is a proxy or not to initialize `Net::HTTP`, according to documentation the default is nil anyway.
http://ruby-doc.org/stdlib-2.0.0/libdoc/net/http/rdoc/Net/HTTP.html#method-c-new